lib/rubocop/cops_documentation_generator.rb

```diff
@@ -0,0 +1,290 @@
+# frozen_string_literal: true
+
+# Class for generating documentation of all cops departments
+class CopsDocumentationGenerator # rubocop:disable Metrics/ClassLength
+  # This class will only generate documentation for cops that belong to one of
+  # the departments given in the `departments` array. E.g. if we only wanted
+  # documentation for Lint cops:
+  #
+  #   CopsDocumentationGenerator.new(departments: ['Lint']).call
+  #
+  def initialize(departments: [])
+    @departments = departments.map(&:to_sym).sort!
+    all_cops = RuboCop::Cop::Cop.registry
+    qualified_cops = all_cops.select { |cop| qualified_cop?(cop) }
+    @cops = RuboCop::Cop::Registry.new(qualified_cops)
+    @config = RuboCop::ConfigLoader.default_configuration
+  end
+
+  def call
```

Comment:
`generate!`?

Reply:
I always thought that for classes that do one thing, and the classname describes what it does, `#call` is the appropriate method name. It could be `#run` or `#main`, but since `#call` is what Ruby uses for Lambdas/Procs, I tend to use it for these purposes.